contracts/build.rs

```diff
@@ -33,6 +33,24 @@ fn main() {
                 )),
             )
     });
+    generate_contract_with_config("BalancerV2WeightedPoolFactory", |builder| {
```

Comment:
can we remove the contract that we were listening to previously?

Reply:
We still need it to get the pool_id

Reply:
We will need the `Vault` contract still for building swap calldata.